### PR TITLE
Get era by ID

### LINKED ISSUE #46: US: Get era by ID
Create getEra query

Probably not a need for a get all eras query because client can get all eras for a specific user in getUser query eventually.

### PR DESCRIPTION
**Pull request by AngelaGuardia**
### Code Highlights:
 * Adds functionality to get a single era by the era ID
 * Updates the era factory so that it generates a date range for start and end date
### Where should the reviewer start?
 * `spec/requests/graphql/queries/eras/get_single_era_spec.rb`
### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.
### Any background context you want to provide?
- N/A
### Issues:
* Closes #46 
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations: ignoring the 'get_' method violations

### CODE REVIEW

File: app/graphql/types/query_type.rb [hunk 1]
```diff
@@ -15,5 +19,10 @@ def get_users
     def get_user(id:)
       User.find(id)
     end
+
+    # Eras
+    def get_era(id:)
```
Comment (AngelaGuardia) [inline app/graphql/types/query_type.rb:33]: I find writing the argument like this `id:` a bit strange. My instinct was to just have `get_era(id)` does anyone know if that is graphql convention/where it comes from?

File: spec/requests/graphql/queries/eras/get_single_era_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          name
+          startDate
+          endDate
+          color
+          createdAt
+          updatedAt
+        }
+      }
+    GQL
+  end
```
Comment (LHJE) [inline spec/requests/graphql/queries/eras/get_single_era_spec.rb:41]: Looks great!

File: spec/factories/eras.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 FactoryBot.define do
+  date = Faker::Date.between(from: 10.years.ago, to: 1.year.ago)
   factory :era do
     name { Faker::Hipster.sentence }
-    start_date { '2020-12-19 15:37:22' }
-    end_date { '2020-12-19 15:37:22' }
+    start_date { date }
+    end_date { Faker::Date.between(from: date, to: Date.today) }
     color { Faker::Color.hex_color }
```
Comment (LHJE) [inline spec/factories/eras.rb:7]: I like this change!

File: app/graphql/types/query_type.rb (hunk 1 above)
Comment (LHJE) [reply app/graphql/types/query_type.rb:33]: I'm not sure but I was feeling the same way...

**Review by LHJE — APPROVED** [review]
Looks good!

File: spec/factories/eras.rb (hunk 3 above)
Comment (leahriffell) [reply spec/factories/eras.rb:7]: Nice! Does `date` on line 5 give a random date?

**Review by leahriffell — APPROVED** [review]
Looks guuud!

File: spec/factories/eras.rb (hunk 3 above)
Comment (AngelaGuardia) [reply spec/factories/eras.rb:7]: It gives a random date between 10 years ago and 1 year ago. Defined on line 2!

Comment (dhalverson) [reply spec/factories/eras.rb:7]: Nice, didn't know you could do that with dates.

**Review by dhalverson — APPROVED** [review]
Looks good, approving and merging. We also discussed the ID part on a call

**Pull request merged**